File: Matura/pos-survival-kit-shareable/example-project/06_Janssen_animalShelter/src/main/java/janssen/javafx/services/SalesService.java
package janssen.javafx.services;

import janssen.javafx.models.Sales;
import janssen.javafx.utils.Http;

import com.fasterxml.jackson.core.JsonProcessingException;

import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;
import java.util.List;
import java.util.concurrent.CompletableFuture;

public class SalesService {

  public static final String ENDPOINT = "/sales";

  public static CompletableFuture<List<Sales>> listAll() {
    return Http.getCollection(ENDPOINT, Sales.class);
  }

  public static CompletableFuture<Sales> get(long id) {
    return Http.get(ENDPOINT + "/" + URLEncoder.encode("" + id, StandardCharsets.UTF_8).replaceAll("[+]", "%20"), Sales.class);
  }

  public static CompletableFuture<Sales> create(Sales sales) throws JsonProcessingException {
    return Http.post(ENDPOINT, sales, Sales.class);
  }

  public static CompletableFuture<Void> update(long id, Sales sales) throws JsonProcessingException {
    return Http.put(ENDPOINT + "/" + URLEncoder.encode("" + id, StandardCharsets.UTF_8).replaceAll("[+]", "%20"), sales);
  }

  public static CompletableFuture<Void> delete(long id) {
    return Http.delete(ENDPOINT + "/" + URLEncoder.encode("" + id, StandardCharsets.UTF_8).replaceAll("[+]", "%20"));
  }

}

File: Matura/pos-survival-kit-shareable/example-project/06_Janssen_animalShelterApp/app/src/main/java/janssen/android/services/SalesService.kt
package janssen.android.services

import janssen.android.models.Sales
import retrofit2.http.*

interface SalesService {
    @GET("/sales")
    suspend fun listAll(): List<Sales>

    @GET("/sales/{id}")
    suspend fun get(@Path("id") id: Long): Sales

    @POST("/sales")
    suspend fun create(@Body sales: Sales): Sales

    @PUT("/sales/{id}")
    suspend fun update(@Path("id") id: Long, @Body sales: Sales): Sales

    @DELETE("/sales/{id}")
    suspend fun delete(@Path("id") id: Long)
}
